summary.go

```diff
@@ -19,21 +19,19 @@ func newSummary(initialCapacity int) *summary {
 	return s
 }
 
-func (s summary) Len() int {
+func (s *summary) Len() int {
 	return len(s.means)
 }
 
 func (s *summary) Add(key float64, value uint32) error {
-
 	if math.IsNaN(key) {
 		return fmt.Errorf("Key must not be NaN")
 	}
-
 	if value == 0 {
 		return fmt.Errorf("Count must be >0")
 	}
 
-	idx := s.FindInsertionIndex(key)
+	idx := s.findInsertionIndex(key)
```

Comment:
I've unexported this since it is only called from Summary and it makes easier to track public API.

Reply:
:+1: